starfish/imagestack/imagestack.py

```diff
@@ -290,6 +290,131 @@ def xarray(self) -> xr.DataArray:
+        # check if X or Y dimension indexed, if so rescale
+        if self._needs_coords_resacling(indexers):
+            self._rescale_physical_coordinates(indexers, new_coords)
+        return new_coords
+
+    def _rescale_physical_coordinates(self, indexers: dict, new_coords: xr.DataArray) -> None:
+        """Iterates through coordinates array and rescales x, y physical coordinate values
+         based on how the x and y dimensions were indexed
+
+        Parameters
+        ----------
+        indexers : a dictionary of dim:index where index is the value
+        or range to index the dimension
+
+        new_coords: the coordinates xarray to modify
+
+        """
+        for _round in range(new_coords.sizes[Indices.ROUND]):
+            for ch in range(new_coords.sizes[Indices.CH]):
+                for z in range(new_coords.sizes[Indices.Z]):
+                    indices = {
+                        Indices.ROUND: _round,
+                        Indices.CH: ch,
+                        Indices.Z: z
+                    }
+                    xmin, xmax = self._rescale_physical_coordinate(
+                        Coordinates.X,
+                        indices,
+                        indexers[Indices.X.value])
+                    ymin, ymax = self._rescale_physical_coordinate(
+                        Coordinates.Y,
+                        indices,
+                        indexers[Indices.Y.value])
+
+                    new_coords[indices].loc[
+                        dict(physical_coordinate=[
+                            PhysicalCoordinateTypes.X_MIN,
+                            PhysicalCoordinateTypes.X_MAX,
+                            PhysicalCoordinateTypes.Y_MIN,
+                            PhysicalCoordinateTypes.Y_MAX]
+                        )] = [xmin, xmax, ymin, ymax]
+
+    def _rescale_physical_coordinate(self, coord: Coordinates, indices: Mapping[Indices, int],
+                                     key) -> Tuple:
+        """Calculates rescaled coordinates
+
+        Parameters
+        ----------
+        coord: The coordinate to rescale.
+
+        indices: The (Round, Ch, Z) indices that identify which tile's coordinates we're using
+
+        key: a value or range to index on
+
+        """
+        coord_min, coord_max = self.coordinates(indices, coord)
+        dimension_size = self.xarray.sizes[coord.value[0]]
+        physical_pixel_size = (coord_max - coord_min) / dimension_size
+        new_min, new_max = coord_min, coord_max
+        min_pixel_index = key if type(key) is int else key.start
+        max_pixel_index = key if type(key) is int else key.stop
+        if min_pixel_index:
+            # check for negative index
+            min_pixel_index = min_pixel_index + dimension_size if min_pixel_index < 0 \
+                else min_pixel_index
+            new_min = (physical_pixel_size * min_pixel_index) + coord_min
+        if max_pixel_index:
```

Comment:
When would max be None/0?

Reply:
So if the slicing operation if from start to somewhere [:3] or end to somewhere [3:] the python Slice object that comes in has a start or stop attribute of None. So we just wanna keep that coordinate value as is.

Reply:
Cool, please immortalize this in a comment. 😁